enhancements/microshift/microshift-updateability-ostree.md

```diff
@@ -0,0 +1,1091 @@
+it means that a backup for that particular deployment must be kept until it disappears
+from ostree. It results in requirement of keeping backup for each deployment separately
+regardless of the MicroShift version they feature.
+
+Because difference between two deployments might not be MicroShift itself, but applications
+that run on top of MicroShift, MicroShift's backups are tied to ostree deployments rather 
+than MicroShift versions.
+
+Decision whether to backup or restore is based on file persisted during previous boot
+(see "Integration with greenboot").
+That file will be removed after successful operation of backup or restore.
+
+There might happen situations when backing up data fails, which will block start of MicroShift
+and result in redboot. It might that after reboot, there might be no backup to restore from.
+**What should we do, before starting MicroShift, in such case is an open question**:
+- Delete the data directory
+  - It case result in loss of MicroShift data, which might be problematic
+    if application used PVC and is given a new one after static manifests are applied.
+- Keep the data dir
+  - Risking that the problem will continue to happen resulting in manual intervention
+
+As a result of investigation and aiming for simplicity for initial implementation,
+it was decided that backing up MicroShift's data will be done by leveraging using 
+copy-on-write (CoW) functionality.
+
+CoW is a feature of filesystem (supported by XFS and Btrfs) and it can be used by 
+providing a `--reflink=` param to `cp` option.
+`--reflink=auto` will be used over `--reflink=always` to gracefully fall back to regular
+copying on filesystems not supporting CoW (ext4, ZFS).
+Since CoW is backed by filesystem, it works only within that filesystem.
+
+To keep track of which backup is intended for which deployment, backup will be placed in a
+directory named after ostree deployment ID inside `/var/lib/microshift.bak/` dir, e.g.
+`/var/lib/microshift.bak/rhel-8497faf62210000ffb5274c8fb159512fd6b9074857ad46820daa1980842d889.0`.
+
+If MicroShift starts and notices that data directory exists, metadata doesn't exist,
+`next-boot` file is missing, it will be assumed that it's a 4.13 to 4.14 upgrade.
+In such case ID of rollback deployment will be used but the automatic rollback will not
+work as 4.13 will not include restore capabilities.
+
+Additionally, a symlink `/var/lib/microshift.bak/latest` will point to the most recent
+backup. This will allow redboots of staged deployment to restore backup of previous
+deployment to use as a starting point (to protect and retry in case of failed data
+migration).
+
+Restore operation works the same, just in the other direction - copying contents of 
+`/var/lib/microshift.bak/ostree-deploy-id/` to `/var/lib/microshift/`.
+
+End user documentation needs to include guidance on setting up filesystem to fullfil 
+requirements for using copy-on-write (e.g. making sure some filesystem options are not disabled).
```

Comment:
CoW is an optimization for saving disk space. Even if the file system does not have the features enabled, we will create a full copy of backup as a fallback, right? If so, let's mention it here explicitly